qiskit/quantum_info/states/statevector.py

```diff
@@ -856,6 +862,21 @@ def _evolve_instruction(statevec, obj, qargs=None):
             return statevec
         if isinstance(obj, Barrier):
             return statevec
+        if isinstance(obj, Initialize):
+            # state is initialized to labels in the initialize object
+            if all(isinstance(param, str) for param in obj.params):
+                statevec._data = Statevector.from_label("".join(obj.params))._data
+            # state is initialized to an integer
+            # here we're only checking the length as (1) a length-1 object necessarily means the
+            # state is described by an integer (as labels were already covered) and (2) the int
+            # was cast to a complex and we cannot do an int typecheck anyways
```

Comment:
wtf is going on with parameter handling in Terra...  Thanks for the comment explaining this, because I'd have had no idea why we were doing this without it.